app/analysis/index/controller.js

```diff
@@ -453,13 +394,20 @@ export default class AnalysisController extends Controller.extend(
 
   @action
   setModelFilter(key, value) {
-    set(this, key, value && value.id);
+    this[key] = value && value.id;
+    this._reset();
   }
 
   @action
   reset() {
-    this.resetQueryParams(
-      Object.keys(this.allQueryParams).filter((k) => k !== "ordering")
+    resetQueryParams(
+      this,
+      Object.keys(allQueryParams(this)).filter((k) => k !== "ordering")
     );
   }
+
+  // i am puting it here in a getter, becuase it's used in the template
```

Comment:
Comment not needed. I think this is obvious and it has a typo in it. :) 

After that we can merge it! :+1: